### PR TITLE
add maker functions for reducers and splitters

### PR DESCRIPTION
**Pull request by hsorsky**
This PR proposes to add functions that return reducer and splitter functions rather than having to write them out yourself (i.e. for delimiters other than "dot" and "underscore")

### CODE REVIEW

**Comment by ianlini** [comment]
Thanks for your contribution!
This is exactly what I thought in #16, though I would use a different approach. Anyway, your approach is quite good for an initial design. We just need to change some style related things, and then I will merge this PR and release a new version.

File: src/flatten_dict/reducer.py [hunk 1]
```diff
@@ -25,3 +25,19 @@ def underscore_reducer(k1, k2):
         return k2
     else:
         return "{0}_{1}".format(k1, k2)
+
+
+def make_reducer(delim):
```
Comment (ianlini) [inline src/flatten_dict/reducer.py:30]: I prefer not using abbreviation here. (delim -> delimiter)

File: src/flatten_dict/reducer.py [hunk 2]
```diff
@@ -25,3 +25,19 @@ def underscore_reducer(k1, k2):
...
+    """Create a reducer with a custom delimiter.
+
+    :param delim: delimiter to use to join keys.
```
Comment (ianlini) [inline src/flatten_dict/reducer.py:33]: I actually use numpy-style docstring, so this should be:

```
"""Create a reducer with a custom delimiter.

Parameters
----------
delimiter : str
    Delimiter to use to join keys.

Returns
-------
f : callable
    Callable that can be passed to `flatten()`'s `reducer` argument.
"""
```

**Comment by hsorsky** [comment]
hopefully the changes address your requests correctly

**Pull request merged**